server/services/gameplay.go

```diff
@@ -26,26 +26,61 @@ func convertMoveRequestInward(req *pb.MoveRequest) *controller.MoveRequest {
 	return domMoveRequest
 }
 
-func convertMoveResponseOutward(res *controller.MoveResponse) *pb.MoveResponse {
+func convertMoveResponseOutward(res controller.MoveResponse) *pb.MoveResponse {
 	pbMoveResponse := pb.MoveResponse{}
 	pbMoveResponse.MoveSuccess = res.MoveSuccess
 	pbMoveResponse.Message = &res.Message
 
 	return &pbMoveResponse
 }
 
+func convertBoardSubscriptionRequestInward(req *pb.BoardSubscriptionRequest) controller.BoardSubscriptionRequest {
+	domBoardSubscriptionRequest := controller.BoardSubscriptionRequest{
+		GameID: req.GetGameId(),
+	}
+
+	return domBoardSubscriptionRequest
+}
+
+func convertBoardUpdateOutward(res controller.BoardUpdate) *pb.BoardUpdate {
+	pbBoardUpdate := pb.BoardUpdate{}
+
+	pbBoard := pb.BoardState{
+		Board: make([]pb.Tile, 32),
+	}
+	for _, t := range res.BoardState.Board {
+		pbBoard.Board = append(pbBoard.Board, pb.Tile(int32(t)))
+	}
+	pbBoardUpdate.BoardState = &pbBoard
+
+	pbMoves := make([]*pb.Move, len(res.PrevMoveSet))
+	for _, m := range res.PrevMoveSet {
+		tempMove := pb.Move{
+			IndexFrom: m.IndexFrom,
+			IndexTo:   m.IndexTo,
+		}
+		pbMoves = append(pbMoves, &tempMove)
+	}
+	pbBoardUpdate.PrevMoveSet = pbMoves
+	return &pbBoardUpdate
+}
+
 // MakeMoves (context.Context, *MoveRequest) (*MoveResponse, error)
 func (s *GameplayServer) MakeMoves(ctx context.Context, req *pb.MoveRequest) (*pb.MoveResponse, error) {
 	domMoveRequest := convertMoveRequestInward(req)
-	// channels may not be best here since only expecting 1 output
-	// I don't think channels are necessary here
-	domMoveResponse := controller.HandleMakeMoves(ctx, domMoveRequest)
+	domMoveResponse, err := controller.HandleMakeMoves(ctx, domMoveRequest)
 
-	return convertMoveResponseOutward(domMoveResponse), nil
+	return convertMoveResponseOutward(domMoveResponse), err
```

Comment:
I _think_ you're supposed to do an early return with `nil, err` if err is non-nil, but can you look that up?

Reply:
Done!